rust/src/main.rs

```diff
@@ -73,17 +72,86 @@ fn main() -> anyhow::Result<()> {
         }
     }));
 
+    #[cfg(any(
+        target_os = "openbsd",
+        target_os = "netbsd",
+        target_os = "freebsd",
+        target_os = "dragonfly",
+        target_os = "macos",
+        target_os = "ios",
+        target_os = "watchos",
+    ))]
+    let enabled_flags: HashSet<_> = config
+        .features
+        .file_flags
+        .unwrap_or_default()
+        .iter()
+        .cloned()
+        .collect();
+
     for test_case in TEST_CASES {
-        if let Some(sc) = &test_case.syscall {
-            if !enabled_features.contains(sc) {
-                println!(
-                    "skipped {}: please add it to the configuration file if you want to run it",
-                    sc
-                );
-                continue;
+        //TODO: There's probably a better way to do this...
+        let mut should_skip = false;
+
+        let mut message = None;
+
+        let features = test_case
+            .required_features
+            .iter()
+            .cloned()
+            .collect::<HashSet<_>>();
+        let missing_features = features.difference(&enabled_features);
+        if missing_features.clone().count() > 0 {
+            should_skip = true;
+
+            let message = message.get_or_insert(String::new());
+            *message += "please add the following features to your configuration:\n";
```

Comment:
This message will be printed if a test case must be skipped because it requires some feature not supported on the target platform, right?  If so, the message is a little misleading, because there's no way for the user to add the feature.

Reply:
Hmm, that's right, I haven't thought of it! Since I haven't implemented features "guessing", I was thinking of printing the message to give a hint on how to enable it with the configuration file, but that's assuming the feature is effectively supported for this file system.

Reply:
How about just "skipped: requires FEATURE"?  That way the user can decide for himself whether it's possible to enable the feature.